### DISCUSSION TITLE
vips text width not work?

### DISCUSSION THREAD

**Discussion by zhifengzhuang** [libvips/libvips#2359] [Q&A]
I try to limit result image width.
`vips text out.png "testtesttesttesttesttesttest" --width 100 --align centre`
However, the result image is 161x9.

vips version is
`vips-8.11.1-Tue Jun 29 07:34:42 UTC 2021`

**Comment by jcupitt** [comment]
Hi @781545872, yes, `width` sets the line length for text wrapping, but if one word is larger than that, it can overflow. It can also make images smaller than `width`.

If you must have an exact size, you'll need to crop and perhaps expand the output of text.

**Reply by zhifengzhuang** [reply]
I'm confused about the using of width. I used to limit width of result using width.It will have a new line if text is longer than width. However, it doesn't work with the newest libvips version.

**Reply by jcupitt** [reply]
It should work. I see:

```
$ vips text x.png "hello world"
$ vipsheader x.png
x.png: 63x10 uchar, 1 band, b-w, pngload
$ vips text x2.png "hello world" --width 50
$ vipsheader x2.png
x.png: 32x25 uchar, 1 band, b-w, pngload
```

Making:

![x](https://user-images.githubusercontent.com/580843/126735646-d7485c54-1f05-4e42-855b-12a6edaf3a56.png)
![x2](https://user-images.githubusercontent.com/580843/126735648-4437f2a9-d61c-4aa1-99dd-3affb723a81d.png)

So `width` has made it split the line at 50 pixels.

**Reply by zhifengzhuang** [reply]
Oh, I see.If the text has no space, it will not split?

**Reply by jcupitt** [reply]
That's right.

You can put U+00AD (SOFT HYPHEN) in the text to mark line break points, if that would help. They are normally invisible, but pango will split the line at that point if it has to. 

You can still get lines larger than `width` of course. For example:

```
$ vips text x.png A --width 50 --dpi 10000
$ vipsheader x.png
x.png: 1114x1216 uchar, 1 band, b-w, pngload
```

One very, very large letter :)

**Reply by zhifengzhuang** [reply]
Thanks for your reply!
